File: app/lib/models/jogo.dart
class Jogo {
  final int id;
  final String nome;
  final String genero;
  final String plataforma;
  final double avaliacao;
  final String descricao;
  final String urlImagem;

  Jogo({
    required this.id,
    required this.nome,
    required this.genero,
    required this.plataforma,
    required this.avaliacao,
    required this.descricao,
    required this.urlImagem,
  });

  factory Jogo.fromJson(Map<String, dynamic> json) {
    return Jogo(
      id: json['id'],
      nome: json['nome'],
      genero: json['genero'],
      plataforma: json['plataforma'],
      avaliacao: (json['avaliacao'] as num).toDouble(),
      descricao: json['descricao'],
      urlImagem: json['urlImagem'],
    );
  }
}

File: src/main/java/com/example/jogos/entity/Jogo.java
package com.example.jogos.entity;

import jakarta.persistence.*;

@Entity
public class Jogo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;
    private String genero;
    private String plataforma;
    private Double avaliacao;
    private String descricao;
    private String urlImagem;

    // Getters e setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }

    public String getGenero() { return genero; }
    public void setGenero(String genero) { this.genero = genero; }

    public String getPlataforma() { return plataforma; }
    public void setPlataforma(String plataforma) { this.plataforma = plataforma; }

    public Double getAvaliacao() { return avaliacao; }
    public void setAvaliacao(Double avaliacao) { this.avaliacao = avaliacao; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }

    public String getUrlImagem() { return urlImagem; }
    public void setUrlImagem(String urlImagem) { this.urlImagem = urlImagem; }
}
